### PR TITLE
add: filesystem write configurations

### PR DESCRIPTION
**Pull request by harshit2283**
Adds 3 configurable options
- filename.prefix -> Allows you to set a custom prefix to start your file's name with. i.e arroyo-{uuid}.parquet
- filename.suffix -> Allows you to overwrite the file's extension, sometimes to identify compression you might use `filename.zstd.parquet`. This will allow you to configure that
- filename.strategy -> This ensures files aren't overwritten when using rebuilding pipelines, there might be a chance of collision if files are being written at a static non - partitioned path.

### CODE REVIEW

File: arroyo-worker/src/connectors/filesystem/local.rs [hunk 1]
```diff
@@ -70,30 +85,44 @@ impl<K: Key, D: Data + Sync + Serialize, V: LocalWriter<D>> LocalFileSystemWrite
...
         }
     }
 
     fn get_or_insert_writer(&mut self, partition: &Option<String>) -> &mut V {
+        let file_suffix = if self.filenaming.filename_suffix.is_some() {
+            self.filenaming.filename_suffix.as_ref().unwrap()
+        } else {
+            V::file_suffix()
+        };
+        let filename_strategy = if self.filename_strategy == FilenameStrategy::Uuid {
```
Comment (jacksonrnewhouse) [inline arroyo-worker/src/connectors/filesystem/local.rs:99]: What about calling this something like `filename_base`?

**Review by jacksonrnewhouse — CHANGES_REQUESTED** [review]
Only must-have is it working on the multipart writer.

File: arroyo-worker/src/connectors/filesystem/local.rs [hunk 2]
```diff
@@ -36,6 +37,8 @@ pub struct LocalFileSystemWriter<K: Key, D: Data + Sync, V: LocalWriter<D>> {
     table_properties: FileSystemTable,
     commit_state: CommitState,
     phantom: PhantomData<(K, D)>,
+    filenaming: Filenaming,
```
[the remark this answers is not in the document]
Comment (harshit2283) [reply arroyo-worker/src/connectors/filesystem/local.rs:40]: Makes sense, have kept a single struct

File: arroyo-worker/src/connectors/filesystem/local.rs [hunk 3]
```diff
@@ -70,30 +85,44 @@ impl<K: Key, D: Data + Sync + Serialize, V: LocalWriter<D>> LocalFileSystemWrite
             table_properties,
             commit_state,
             phantom: PhantomData,
+            filenaming,
+            filename_strategy,
         }
     }
 
     fn get_or_insert_writer(&mut self, partition: &Option<String>) -> &mut V {
```
[the remark this answers is not in the document]
Comment (harshit2283) [reply arroyo-worker/src/connectors/filesystem/local.rs:86]: I've added redundant code for now, will refactor and unify.

**Review by jacksonrnewhouse — APPROVED** [review]

**Pull request merged**